trane/core/utils.py

```diff
@@ -164,14 +164,32 @@ def _generate_possible_operations(
         transformation_operations,
         filter_operations,
     ):
+        if (
+            len(
+                {
+                    agg_operation.__name__,
+                    transform_operation.__name__,
+                    filter_operation.__name__,
+                }.intersection(
+                    agg_operation.restricted_ops.union(
```

Comment:
restricted_ops could be converted to the actual python objects, rather than relying on strings

Reply:
I had circular dependency issues when I tried this (since in agg ops I am restricting transformation ops, and in transformation ops I am restricting agg ops; so I need to import one into another and vice versa), do you have an idea on how to approach it?

Reply:
I definitely agree with this though

Reply:
oh I see. Let's leave it for now that, as is